snapcraft_parser/tests/test_parser.py

```diff
@@ -0,0 +1,284 @@
+        }
+        main(["--debug", "--index", TEST_OUTPUT_PATH])
+        self.assertEqual(0, _get_part_list_count())
+
+    @mock.patch('snapcraft_parser.main._get_origin_data')
+    @mock.patch('snapcraft.internal.sources.get')
+    def test_single_part_origin(self,
+                                mock_get,
+                                mock_get_origin_data):
+        """Test a wiki entry with a single origin part."""
+        _create_example_output("""
+example:
+    maintainer: John Doe <john.doe@example.com
+    origin: lp:snapcraft-parser-example
+    description: example
+    project-part: main
+""")
+        mock_get_origin_data.return_value = {
+            "parts": {
+                "main": {
+                    "source": "lp:something",
+                    "plugin": "copy",
+                    "files": ["file1", "file2"],
+                },
+            }
+        }
+        main(["--debug", "--index", TEST_OUTPUT_PATH])
+
+        self.assertEqual(1, _get_part_list_count())
+
+    @mock.patch('snapcraft_parser.main._get_origin_data')
+    @mock.patch('snapcraft.internal.sources.get')
+    def test_multiple_part_origin(self,
+                                  mock_get,
+                                  mock_get_origin_data):
+        """Test a wiki entry with multiple origin parts."""
+        _create_example_output("""
+example:
+    maintainer: John Doe <john.doe@example.com
+    origin: lp:snapcraft-parser-example
+    description: example
+    project-part: main
+    parts: ['subpart']
+""")
+        mock_get_origin_data.return_value = {
+            "parts": {
+                "main": {
+                    "source": "lp:something",
+                    "plugin": "copy",
+                    "files": ["file1", "file2"],
+                    "after": ["subpart"],
+                },
+                "subpart": {
+                    "source": "lp:somethingelse",
+                    "plugin": "copy",
+                    "files": ["subfile2"],
+                },
+            }
+        }
+        main(["--debug", "--index", TEST_OUTPUT_PATH])
+
+        self.assertEqual(2, _get_part_list_count())
+
+    @mock.patch('snapcraft_parser.main._get_origin_data')
+    @mock.patch('snapcraft.internal.sources.get')
+    def test_output_parameter(self,
+                              mock_get,
+                              mock_get_origin_data):
+        """Test a wiki entry with multiple origin parts."""
+        _create_example_output("""
+example:
+    maintainer: John Doe <john.doe@example.com
+    origin: lp:snapcraft-parser-example
+    description: example
+    project-part: main
+    parts: []
+""")
+        mock_get_origin_data.return_value = {
+            "parts": {
+                "main": {
+                    "source": "lp:something",
+                    "plugin": "copy",
+                    "files": ["file1", "file2"],
+                },
+            }
+        }
+
+        filename = "parts.gz"
+        try:
```

Comment:
the `try` `except` here should not be needed, these tests are always run in a new tempdir for each test as part of the fixture setup.